src/resolve/resolved-manager.c

```diff
@@ -1200,7 +1200,7 @@ int manager_compile_dns_servers(Manager *m, OrderedSet **dns) {
         return 0;
 }
 
-int manager_compile_search_domains(Manager *m, OrderedSet **domains, int filter_route) {
+int manager_compile_search_domains(Manager *m, OrderedSet **domains, bool filter_route) {
```

Comment:
so it appears we only invoke manager_compile_search_domains() a single time, and then we want to filter out the routing domains. maybe we should just get rid of the "filter_route" parameter altogether and unconditionally filter out route_only domains here. (Though this should get an explicit comment, so that that people who read the code won't be surprised)